data-access-layer/iaas/GcpClient.js

```diff
@@ -30,6 +30,76 @@ class GcpClient extends BaseCloudClient {
     );
   }
 
+  getDiskMetadata(diskCid, zone) {
+    return Promise.try(() =>
+      this.computeClient
+      .zone(zone)
+      .disk(diskCid)
+      .get()
+      .then(diskData => diskData[0].metadata)
+      .then(metadata => {
+        return {
+          volumeId: metadata.name,
+          size: metadata.sizeGb,
+          zone: metadata.zone.substring(metadata.zone.lastIndexOf('/') + 1),
+          type: metadata.type.substring(metadata.type.lastIndexOf('/') + 1),
+          extra: {
+            tags: metadata.labels,
+            type: metadata.type
+          }
+        };
+      })
+    );
+  }
+
+  createDiskFromSnapshot(snapshotId, zone, opts = {}) {
+    return new Promise((resolve, reject) => {
+      const diskName = this.getRandomDiskId();
+      const options = {
+        sourceSnapshot: `global/snapshots/${snapshotId}`,
+        description: 'disk created via service fabrik for bosh restore',
```

Comment:
May be we can remove ` for bosh restore`  as same function might be used in future for different operator's instances other than BOSH. May be for container data restore also we use the same method.